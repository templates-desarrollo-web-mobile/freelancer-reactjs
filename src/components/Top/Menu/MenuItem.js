import PropTypes from "prop-types";

const MenuItem = ({ item }) => (
  <li className="nav-item mx-0 mx-lg-1">
    <a className="nav-link py-3 px-0 px-lg-3 rounded" href={"#".concat(item.id)}>
      {item.title}
    </a>
  </li>
);

export default MenuItem;

MenuItem.propTypes = {
    item: PropTypes.object.isRequired,
};