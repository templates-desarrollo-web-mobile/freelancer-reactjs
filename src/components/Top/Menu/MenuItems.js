import PropTypes from "prop-types";
import MenuItem from "./MenuItem";

const MenuItems = ({items}) => (
  <div className="collapse navbar-collapse" id="navbarResponsive">
    <ul className="navbar-nav ms-auto">
        {
          items.map((item) => <MenuItem key={item.id} item={item}/>)
      }
    </ul>
  </div>
);


export default MenuItems;

// Para obligar pasar el argumento para la funci√≥n
MenuItems.propTypes = {
    items: PropTypes.array.isRequired,
};