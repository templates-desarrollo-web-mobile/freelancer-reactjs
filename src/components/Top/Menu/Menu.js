import { menuItems } from "../../../helpers/menuItems";
import MenuItems from "./MenuItems";

const Menu = () => {
  return (
    <nav
      className="navbar navbar-expand-lg bg-secondary text-uppercase fixed-top"
      id="mainNav"
    >
      <div className="container">
        <a className="navbar-brand" href="#page-top">
          Start Bootstrap
        </a>
        <button
          className="navbar-toggler text-uppercase font-weight-bold bg-primary text-white rounded"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarResponsive"
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          Menu
          <i className="fas fa-bars"></i>
        </button>
        <MenuItems items={menuItems} />
      </div>
    </nav>
  );
};

export default Menu;
