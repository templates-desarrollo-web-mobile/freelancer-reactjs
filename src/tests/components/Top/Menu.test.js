import { shallow } from 'enzyme';
import Menu from "../../../components/Top/Menu/Menu";
describe('Pruebas en "Footer"', () => {
    
    const wrapper = shallow(<Menu/>);
    test('Debe de mostrar correctamente el componente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

});