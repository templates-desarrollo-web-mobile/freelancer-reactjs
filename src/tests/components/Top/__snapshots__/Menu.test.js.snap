// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en "Footer" Debe de mostrar correctamente el componente 1`] = `
<nav
  className="navbar navbar-expand-lg bg-secondary text-uppercase fixed-top"
  id="mainNav"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#page-top"
    >
      Start Bootstrap
    </a>
    <button
      aria-controls="navbarResponsive"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler text-uppercase font-weight-bold bg-primary text-white rounded"
      data-bs-target="#navbarResponsive"
      data-bs-toggle="collapse"
      type="button"
    >
      Menu
      <i
        className="fas fa-bars"
      />
    </button>
    <MenuItems
      items={
        Array [
          Object {
            "id": "portfolio",
            "title": "Portfolio",
          },
          Object {
            "id": "about",
            "title": "About",
          },
          Object {
            "id": "contact",
            "title": "Contact",
          },
        ]
      }
    />
  </div>
</nav>
`;
